helix-term/src/commands.rs

```diff
@@ -2992,14 +2993,36 @@ pub mod insert {
                         (start, pos, None) // delete!
                     }
                 } else {
-                    // delete char
-                    (
-                        graphemes::nth_prev_grapheme_boundary(text, pos, count),
-                        pos,
-                        None,
-                    )
+                    match (
+                        text.get_char(pos.saturating_sub(1)),
+                        text.get_char(pos),
+                        auto_pairs,
+                    ) {
+                        (Some(_x), Some(_y), Some(ap))
+                            if range.len() == 1
```

Comment:
Might be better to use [`Range::is_single_grapheme`](https://github.com/helix-editor/helix/blob/9f43dbc45d254ff860f7d33db201b2f0a6073195/helix-core/src/selection.rs#L344) so that it also works when the cursor is over a non-ASCII grapheme.